apps/enc.c

```diff
@@ -476,9 +476,13 @@ int enc_main(int argc, char **argv)
             BIO_printf(bio_err, "iv undefined\n");
             goto end;
         }
-        if ((hkey != NULL) && !set_hex(hkey, key, EVP_CIPHER_key_length(cipher))) {
-            BIO_printf(bio_err, "invalid hex key value\n");
-            goto end;
+        if (hkey != NULL) {
+            if( !set_hex(hkey, key, EVP_CIPHER_key_length(cipher))) {
```

Comment:
Formatting nit, it's `if (!`, not `if( !`.

Reply:
Sorry . Done.